src/main/java/eu/bbmri/eric/csit/service/negotiator/configuration/SecurityConfig.java

```diff
@@ -73,20 +77,17 @@ protected void configure(HttpSecurity http) throws Exception {
         .disable();
 
     http.authorizeRequests()
-        .antMatchers(HttpMethod.POST, "/v3/data-sources/**")
+        .antMatchers(HttpMethod.POST, "/v3/data-sources/*")
         .hasAuthority("ADMIN")
         .and()
-
         .authorizeRequests()
-        .antMatchers("/v3/projects/**", "/v3/negotiations/**", "/v3/access-criteria")
+        .antMatchers("/v3/projects/**", "/v3/negotiations/**", "/v3/access-criteria/**")
         .hasAuthority("RESEARCHER")
         .and()
-
         .authorizeRequests()
-        .antMatchers(HttpMethod.POST, "/v3/requests/**")
+        .antMatchers(HttpMethod.POST, "/v3/requests/**", "/directory/create_query")
```

Comment:
shouldn't this be v2/directory/create_query?

Reply:
no, because the old api didn't have the version prefix. for example the directory calls /api/directory/create_query